WANT_VST_NOWINE addition to CMakeLists

Israel- opened the pull request:
Here you go Tres!

Wallacoloo commented:
This PR comes from the discussion of #2132, for those lacking the context.

Looks fine to me, but I'll let @tresf do the merging since he is more on top of the cmake stuff ;-)

Israel- commented:
thanks @Wallacoloo for the additional info added... I was sparse in my excitement to finally get to this :)

Israel- commented:
I added in the debian skelton directory and added some stuff to gitignore to ignore all the CMake stuff generated by running:
```cmake .```
I didn't modify the CMakeLists.txt yet as I want someone like @falkTX to look through this and see if this works... the CMakeLists for the debian folder is there...
feedback appreciated

tresf commented:
@Israel- thanks kindly.

First, we just cleaned up the root directory tree, can we move these items into cmake/debian please?  Variable scope can get very tricky with CMake, so we'll need to work through this.

Second, can you explain what the separate directories are for?  i.e. `/debian-conf` vs. `/debian/source`, etc?

Last, CMake has the capacity to fire a `make package` directly into a debian file.  I assume your solutions is still manual (that is fine, it is manual on OSX too) but we may want to put some `FIXME:` tags in the source if we're going to skip the built-in packaging support so that other devs are well aware we're rolling our own.

http://www.cmake.org/Wiki/CMake:CPackPackageGenerators#DEB_.28UNIX_only.29

-Tres

Israel- commented:
oh, sure thing.
debian is the real debian directory with the static files... though they could be combined I guess.... I just did it to separate things to make it clear what is going on here, though it might have had the opposite effect. :(

debian-conf is the *.in files.
I did this to try to make things more clear, but if it is less clear we can merge the two folders, easily.
I think it would be better to leave it in the source directory, as when we package it, this folder goes in the root directory.
debian/source directory is a directory containing the format file.
I still need to add a few things (like compat, and editing the top-level CMakeLists.txt) to make it fully functional.  I would like feedback from @falkTX if possible.  I don't want this to become a problematic issue for anyone or any distro.

As far as CPack stuff goes, that is probably a great solution, to some degree... however I do not think we can fully implement the VST solutions this way, since we have greatly customized setup.  Either way, I will modify the variables to use CPack variable names for future use.... then we can decide where to put things, and test everything out :)

tresf commented:
> I think it would be better to leave it in the source directory

What is `source` directory? Do you mean `/src`?  If you look at `cmake/linux`, `cmake/nsis`, `cmake/apple`, you'll notice I've pulled all of the platform-specific stuff out of the root because it clutters our directory tree and is almost all related to installing or packaging.

Hypothetically, developers won't ever package as they're focusing on fixing bugs and may even be running multiple versions.  The packaging stuff should be separated or else our `/CMakeLists.txt` will grow a mile long (which is already is, BTW, but we're shortening this as we progress).

> when we package it, this folder goes in the root directory.

So do most of our items in `cmake/<foo>`.  The destination isn't that relevant yet as we have no standards around how we organize our packing helpers.  Where it ends up is mostly moot too, since packaging only happens once per release, so if there's clutter left behind, only the packager suffers, and they're less likely to care so as long as the package is clean.

This can be observed by the Windows packager, which uses NSIS for packaging.  There are files all over the `/build` directory such as RC files for setting up EXE resources, etc but no one cares because Windows is intended to be run from package rather than from the `/build` directory.  For now, I vote we keep `.deb` packaging the same.  We just drop it in `/build` and let the packager do the rest.

> As far as CPack stuff goes, that is probably a great solution, to some degree... however I do not think we can fully implement the VST solutions this way, since we have greatly customized setup

This is the same problem we encounter with Mac.  Mac has linking errors all over the place and I patch these with a shell script and some Mac-linker tools.  Don't worry much about the CPack variables unless we need them as they'll only confuse us down the road.  On the contrary if you learn CPack can do what we need, it will allow us to use `make package` properly on that platform, so it is worth trying, but not worth killing yourself over.

> I still need to add a few things (like compat, and editing the top-level CMakeLists.txt) to make it fully functional. I would like feedback from @falkTX if possible. I don't want this to become a problematic issue for anyone or any distro.

Variable scope in CMake using `PARENT_SCOPE` can allow us to modify the top-level `CMakeLists.txt` without bulking up the file.  This is very tricky tho so I'll help, but this is what I'm doing with MSYS support via [`#2205`](https://github.com/LMMS/lmms/pull/2205/files#diff-0d86ce52f30a2e7bb7ec7cea1b0571e3R51).  It is ugly and difficult to debug, but it works well to isolate the one-offs from our root `CMakeLists.txt`. Unless we have a technical reason to break this, is the only approach I'll work with as it brings organization to our bundling which was previously spammed into our root `CMakeLists.txt`.  I think it help contain the scope of our platform (and distro) dependent logic without confusing someone looking to turn off `CALF`, `ZYN`, etc.

Israel- commented:
Ok,
I think I will move the debian folder into the cmake/linux folder and move the desktop.in file there and rework things.
I am still going to use the cpack variables in the debian directory CMakeLists.txt as this will be a quick way to convert everything IF we can ever use CPack (like if it gets more variables and customization options).  If we keep all Linux related stuff in the cmake/linux directory it should be clear to people what is going on.  I simply did not know this existed.
I will clean things up and work this out

tresf commented:
Great.  Keep context in mind though... `lmms.desktop.in` is a Linux file (it conforms to the Linux desktop shortcut standard), not necessarily a Debian-specific file. :+1: 

-Tres

Israel- commented:
right I am moving the menu.in and desktop.in files into the linux directory.
I think this will be a good overall solution to things.
If a packager wants to have the debian folder generated automatically they can run
cmake .
And if they want to 'roll their own' it is a simple as not running ```cmake .```
Is this acceptable way to do this?

tresf commented:
Custom targets are the way I'm managing it on Apple now.  That way anyone can call `make dmg` or in your case, perhaps, `make deb` or `make dpkg` or whatever until the `CPack` logic is finalized.

https://github.com/LMMS/lmms/pull/2271/files#diff-764d81bb8cb9b48be475503ac96268a5R23

tresf commented:
So to answer your question... Linux specific files can go straight into the `cmake/linux/CMakeLists.txt`, we're already do platform detection to invoke that.

If Debian needs its own files that other Linuxes won't need, we'll do more granular detection in `cmake/linux/CMakeLists.txt` prior to including the debian directory (usually reading `lsb-release` is a good way to detect a debian flavor).

-Tres

Israel- commented:
Tres, can you write me a quick check for this?  I only work on Debian based packaging, so I have never needed to come up with something like this.  Or you can point me to documentation on it.  I am getting fairly close to working the debian portion out

Israel- commented:
is it as simple as:
```
find_program(LSB lsb_release)
if (LSB)
MESSAGE("Found lsb_release")
ADD_SUBDIRECTORY(debian)
endif (LSB)
```
or am I way off here..

tresf commented:
> Tres, can you write me a quick check for this?

Can you try this out?

Please put it in the file `cmake/modules/DetectMachine.cmake`
```cmake
# [...]
ENDIF(WIN32)

############## THIS PART ###############

# Detect Debian, Ubuntu and family
IF(EXISTS /etc/lsb-release)
        SET(LMMS_BUILD_DEBIAN 1)
# Detect RedHat, Fedora and family
ELSEIF(EXISTS /etc/redhat-release)
        SET(LMMS_BUILD_REDHAT 1)
ENDIF()

```

And then the logic would be:

in `cmake/CMakeLists.txt`

```cmake
ELSE()
       ADD_SUBDIRECTORY(linux)

############## THIS PART ###############
       IF(LMMS_BUILD_DEBIAN)
               ADD_SUBDIRECTORY(debian)
       ENDIF()
ENDIF()
```

... **but** this is where variable scope gets hairy... If you need to expose a variable to `/CMakeLists.txt`, you can't do it from nested subdirectories without a bunch of ugliness.  If you have issues with this approach, you can easily move the conditional to `/CMakeLists.txt` and then `PARENT_SCOPE` will expose variables (such as CPack ones) to the parent.  This whole scope thing is depending on a bunch of order of operations as well as the fact that a child `CMakeLists.txt` can use parent scope, but there exists no such thing as ~~`GRANDPARENT_SCOPE`~~.

Is that confusing?  For NSIS to be nested, I had to do a lot of trial and error with variable scope, but I hope that at least gets your platform detection logic happy. :+1:

Israel- commented:
Check it out and see if you like the new style and version here

tresf commented:
I made all of my comments in-line with code to illustrate context.

Israel- commented:
This code creates the debian directory and files in ${CMAKE_SOURCE_DIR} (the root directory).
debian/watch

tresf commented:
> This code creates the debian directory and files in ${CMAKE_SOURCE_DIR} (the root directory).
debian/watch

But are they needed there?  Shouldn't packaging be done via the `${CMAKE_BINARY_DIR}` ?

Israel- commented:
No, this creates the files to create the debian archive which installs things.  Think of the debian directory as similar to the cmake directory.  It simply controls the build and installation for Debian ONLY.

tresf commented:
> Sorry I suppose these are equivalent in this case... we can switch the variables.

Yeah for people who build from the root of the source directory, our build instructions on the other hand recommend building from a separate directory.  I'll reference `build` from time to time, this is the directory our build instructions say to build from.  We invoke `cmake ..` to allow the built files to be in a special directory, which prevents your `git commit -a` from picking up a bunch of build files that are irrelevant to the source code. :+1:

Israel- commented:
oh... ok well that is nice to know! Here is the fix for that.

tresf commented:
So FYI, variable scope is still going to be a problem... If you intend to use the variable with CPack, it **must** be declared parent scope and it **must** be `ADD_SUBDIRECTORY(...)` from `cmake/CMakeLists.txt` but this will hide the variable from itself (yeah, this is ugly -- blame cmake).  So variables that you'll need to reference within `cmake/linux/debian/CMakeLists.txt` need to be doubly declared as both `PARENT_SCOPE` and current scope like so:

```cmake
SET(CMAKE_CPACK_FOO "bar")
SET(CMAKE_CPACK_FOO "${CMAKE_CPACK_FOO}" PARENT_SCOPE)
```

If you don't plan on using them inside the `debian/CMakeLists.txt`, you can just declare them as parent scope and all is happy.  So anything with `CPACK` in the name must be parent scope and if you need that further down the file, make sure to declare it twice like above.

This is why most of our code has this junk all declared in the top level CMakeLists.txt because the variable scope is confusing.  It is a necessary evil though. :+1:

Israel- commented:
Ok, well for now we are not using CPack, so just having them declared will make it easier if we can get CPack to work we only need to copy the variables to the base directory CMakeLists

tresf commented:
> Ok, well for now we are not using CPack, so just having them declared will make it easier if we can get CPack to work we only need to copy the variables to the base directory CMakeLists

We shouldn't leave confusing breadcrumbs for ourselves later.  I vote we either we leave them out, or we get the `PARENT_SCOPE` stuff worked out now.

tresf commented:
@Israel- can you help clean this up prior to merging?

 1. `cmake/linux/lmms` refers to an `.xpm` icon file.  Where is this?
    * You've also added a reference to this icon in `cmake/linux/CMakeLists.txt`.  Should it be a variable?
 1. What is the `CONTRIBUTORS` file for?  Can we instead use the `CONTRIBUTORS` variable that has already been set in [`CMakeLists.txt#L419`](https://github.com/LMMS/lmms/blob/master/CMakeLists.txt#L419)?  If this needs to be a physical file, can we write it out at build time rather than committing a redundant file into the repo?
 1. The `.gitignore` has a bunch of items that I think were added because the build was being fired in the root of the source tree.  Can you please adjust your build process to build in an alternate location so that we don't need to muck up this file?
 1. Can `ENG_COMMENT` and `ENG_GENERIC` be removed and just use [`PROJECT_DESCRIPTION`](https://github.com/LMMS/lmms/blob/master/CMakeLists.txt#L31) so that we don't have redundant definitions?
 1. In `lmms.desktop` you [removed the actual `lmms` command](https://github.com/LMMS/lmms/pull/2298/files#diff-2f50c0e8621ce625bfe4e155028bd5feL9).  Was this intentional?  Does it still work?  Can you explain?
 1.  I can't comment on too much of the build scripts, but the rest looks ok.  Once we finish these items, we can squash down to a single commit and get this in our master branch for the 1.2 release.

Israel- commented:
Ok @tresf I think I got this fixed up the way you wanted!
I removed the lmms.desktop file completely, but tested running cmake and it generates it correctly in my tests, please check my work and see if this meets approval.  I cleaned up gitignore as well

tresf commented on cmake/linux/debian/CMakeLists.txt:
```diff
@@ -0,0 +1,60 @@
+SET(CPACK_DEBIAN_PACKAGE_NAME "${CMAKE_PROJECT_NAME}")
+SET(CPACK_DEBIAN_PACKAGE_SECTION "sound")
+SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "${PROJECT_AUTHOR} <${PROJECT_EMAIL}>")
+set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "${PROJECT_URL}")
```
why ucase `set` ?

tresf commented on cmake/linux/debian/CMakeLists.txt:
```diff
@@ -0,0 +1,60 @@
+set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "${PROJECT_URL}")
+SET(CPACK_DEBIAN_PACKAGE_ARCHITECTURE "any")
+SET(CPACK_PACKAGE_DESCRIPTION_SUMMARY "${PROJECT_DESCRIPTION}")
+SET(CPACK_DEBIAN_PACKAGE_RECOMMENDS "${CPACK_DEBIAN_PACKAGE_NAME}-vst-full (= ${DOLLAZ}{binary:Version}) [amd64 i386], tap-plugins, caps")
```
What is `${DOLLAZ}` ?

tresf commented on debian/source/format:
```diff
@@ -0,0 +1 @@
+3.0 (native)
```
Does this file have to be in a separate `/source/` folder?

Israel- commented:
```
SET(CPACK_PACKAGE_DESCRIPTION " ${PROJECT_NAME_UCASE} .... ")
```
is to make the CPACK_PACKAGE_DESCRIPTION use LMMS, PROJECT_NAME_UCASE is defined in the root CMakeLists.txt

$DOLLAZ is a literal $  the debian control file ALSO has variable subsitition and I don't know how to escape the literal $ in a CMakeLists.txt any other way... I'm open to suggestions :)

yes format needs to be in its own debian/source directory

tresf commented on cmake/linux/CMakeLists.txt:
```diff
@@ -2,3 +2,16 @@ INSTALL(FILES lmms.png                  DESTINATION "${DATA_DIR}/pixmaps")
 INSTALL(FILES lmms.desktop              DESTINATION "${DATA_DIR}/applications")
 INSTALL(FILES lmms.xml                  DESTINATION "${DATA_DIR}/mime/packages")
+
+SET(EXEC "env QT_X11_NO_NATIVE_MENUBAR=1 lmms")
```
We should follow some basic naming conventions when making variable names... This isn't mandatory, but instead of `EXEC`, we should do something like `LINUX_SHORTCUT_EXEC`, `LINUX_SHORTCUT_ICON`, `LINUX_SHORTCUT_ICON_XPM`.  Also, what is `lmms`?  Is that some automagic icon?  Do we not need to link to the SVG/PNG?  Is the 32_32 a backwards compat thing?  Or is this a 32x32 pixel icon?  Is is always an XPM?

Israel- replied:
lmms is the menu file.... though a recent e-mail on the debian list suggests that soon this will go away since using a desktop file is much more common.
I can change the variables if you'd rather I did

tresf replied:
I was inquiring about the `SET(ICON "lmms")`, not so much the menu portion.

tresf commented on cmake/linux/debian/CMakeLists.txt:
```diff
@@ -0,0 +1,60 @@
+ ${PROJECT_NAME_UCASE} combines the features of a tracker-/sequencer-program (pattern-/channel-/
+ sample-/song-/effect-management) and those of powerful synthesizers and
+ samplers in a modern, user-friendly and easy to use graphical user-interface")
+SET(ICON_INSTALL "data/lmms.png")
```
Like above, not mandatry, but we should use some naming convention here that make sense.  `ICON_INSTALL` is certainly better than `ICON`, but if the name can help explain context, that helps as I've no clue the difference between the different icon variables at a glance.

tresf commented on cmake/linux/debian/lmms-vst-full.install.in:
```diff
@@ -0,0 +1 @@
+usr/lib/lmms/RemoteVstPlugin*
```
It doesn't hurt to keep this as an `.in` file, but just an FYI, if you aren't providing it parameters, it technically doesn't have to be configured.  Feel free to leave this as-is if you feel this may take some CMake params down the road.

Israel- replied:
yeah, if you'd rather I just move the completed file into the base debian directory I will

tresf replied:
Naw, this is fine for now, just an FYI. :+1:

tresf commented:
> $DOLLAZ is a literal $ [...] I don't know how to escape the literal $ in a CMakeLists.txt any other way... I'm open to suggestions :)

Please see [`cmake/apple/CMakeLists.txt#L14`](https://github.com/LMMS/lmms/blob/master/cmake/apple/CMakeLists.txt#L14)

Israel- commented:
How do you do it for something like
```
SET(CPACK_DEBIAN_PACKAGE_RECOMMENDS "${CPACK_DEBIAN_PACKAGE_NAME}-vst-full (= ${DOLLAZ}{binary:Version}) [amd64 i386], tap-plugins, caps")
```

tresf commented on cmake/linux/lmms:
```diff
@@ -1,4 +1,4 @@
   title="LMMS" hints="Audio" command="/usr/bin/lmms" \
   longtitle="LMMS" \
-  icon="/usr/share/pixmaps/lmms.png"
+  icon32x32="/usr/share/pixmaps/lmms.xpm"
```
I'd recommend we hard code as little of this as possible and `CONFIGURE` it.
 * `lmms` in `/usr/bin/lmms` can be assumed
 * Longtitle can be assumed
 * Icon is a dupe of `/usr/share/pixmaps/lmms.xpm` defined in `cmake/linux/CMakeLists.txt`

tresf replied:
Actually, isn't this entire file a duplicate of `cmake/linux/menu.in`?

tresf commented:
> How do you do it for something like
> ```
> SET(CPACK_DEBIAN_PACKAGE_RECOMMENDS "${CPACK_DEBIAN_PACKAGE_NAME}-vst-full 
(= ${DOLLAZ}{binary:Version}) [amd64 i386], tap-plugins, caps")
> ```

Ah I see the problem... escaping them should do the trick. Can you try:

```cmake
SET(FOO "Bar Lorem Ipsum \${do_not_evaluate}")
```

More reading on this available here:  https://www.mail-archive.com/cmake@cmake.org/msg11302.html

Disclaimer, the author ends stating that he's too lazy to update the cmake wiki. :laughing:

tresf commented:
@Israel- this is coming along very nicely.  Thanks for taking the time to clean this up!!!

Israel- commented:
Ok... hopefully this fixes all the issues you have found!
I will try to rebuild this and make sure these fancy escape sequences actually work, I was originally trying to escape both things, but using @ONLY as well as \$ (though that e-mail looks like there are lots of caveats) should work

Israel- commented:
I hope these newer names make a bit more sense
```
LINUX_RELATIVE_DESKTOP_FILE_ICON
DEBIAN_INSTALL_FILE_ICON_LINE
```
DEBIAN_INSTALL_FILE_ICON_LINE is the path in the lmms source where the icon exists, I did this to make any future updates easier to manage (changing a variable instead of figuring out where I hid it )
I hope this makes it much clearer.  If there are any other issues you find I will deal with them on the morrow

tresf commented:
```
LINUX_RELATIVE_DESKTOP_FILE_ICON
DEBIAN_INSTALL_FILE_ICON_LINE
```

Well, I'd prefer them closer to what was proposed in https://github.com/Israel-/lmms/commit/8a3866a4869f533e595d529d817e7194cf5bb6a0#commitcomment-13278773, (notice the shorter names and use of comment when something isn't clear).  Also, I grouped the Linux lines together and put the Menu line afterward.  For example, `LINUX_DESKTOP_ICON` can be assumed to be a file, and `RELATIVE` can be a comment, since I'm not confident this approach will work cross platform (remember, this `lmms.desktop` file gets created in the Linux portion, so all distros are affected -- even if only Debian uses it today)

Also, I'm still not sure why the word `_LINE` in the `DEBIAN_INSTALL_FILE_ICON_LINE` variable name, since all variables eventually get placed on a line somewhere.   But if these are the names you'd like, I won't beat them up any more.

The tab formatting however (well, "space formatting", technically) could [still be improved](https://cloud.githubusercontent.com/assets/6345473/9933347/58b7c8d4-5d18-11e5-9493-5b1bfa62d4a5.png)

tresf commented on cmake/linux/CMakeLists.txt:
```diff
@@ -2,3 +2,9 @@ INSTALL(FILES lmms.png                  DESTINATION "${DATA_DIR}/pixmaps")
+SET(MENU_FILE_ICON_XPM "/usr/share/pixmaps/lmms.xpm")
+SET(LINUX_RELATIVE_DESKTOP_FILE_ICON "lmms")
+CONFIGURE_FILE(lmms.desktop.in ${CMAKE_PROJECT_NAME}.desktop)
+CONFIGURE_FILE(menu.in ${CMAKE_PROJECT_NAME})
```
I don't think we should place a file anywhere that is simply called `LMMS` with no file extension.  Can you link me to a document that says the menu system requires this?

Israel- replied:
we should probably call it lmms.1

tresf replied:
Hopefully not as `lmms.1` is the name of our manpage.  Can you help me understand what this is exactly?  Where in the freedesktop spec does it say to provide an extension-less file?  I'm not trying to disagree, just understand. :+1:

tresf commented on cmake/linux/CMakeLists.txt:
```diff
@@ -2,3 +2,9 @@ INSTALL(FILES lmms.png                  DESTINATION "${DATA_DIR}/pixmaps")
 INSTALL(FILES lmms.desktop              DESTINATION "${DATA_DIR}/applications")
 INSTALL(FILES lmms.xml                  DESTINATION "${DATA_DIR}/mime/packages")
+
+SET(LINUX_DESKTOP_FILE_EXEC "env QT_X11_NO_NATIVE_MENUBAR=1 lmms %f")
```
The word `FILE` is redundant, please rename to `LINUX_DESKTOP_EXEC` here as well as in the desktop file.

Israel- replied:
ok

tresf commented on cmake/linux/CMakeLists.txt:
```diff
@@ -2,3 +2,9 @@ INSTALL(FILES lmms.png                  DESTINATION "${DATA_DIR}/pixmaps")
+
+SET(LINUX_DESKTOP_FILE_EXEC "env QT_X11_NO_NATIVE_MENUBAR=1 lmms %f")
+SET(MENU_FILE_ICON_XPM "/usr/share/pixmaps/lmms.xpm")
+SET(LINUX_RELATIVE_DESKTOP_FILE_ICON "lmms")
```
The word `FILE` is redundant, please rename to `LINUX_DESKTOP_ICON` here as well as in the desktop file.

tresf replied:
Actually on second thought... remove this entirely and use `${CMAKE_PROJECT_NAME}` in the desktop file as this is only used once now that the Debian stuff has been removed. :+1:

tresf commented on cmake/linux/CMakeLists.txt:
```diff
@@ -2,3 +2,9 @@ INSTALL(FILES lmms.png                  DESTINATION "${DATA_DIR}/pixmaps")
 INSTALL(FILES lmms.xml                  DESTINATION "${DATA_DIR}/mime/packages")
+
+SET(LINUX_DESKTOP_FILE_EXEC "env QT_X11_NO_NATIVE_MENUBAR=1 lmms %f")
+SET(MENU_FILE_ICON_XPM "/usr/share/pixmaps/lmms.xpm")
```
Is the menu something that was forgotten when removing the debian stuff or is this good for all Linuxes?  If we keep it, I would recommend simply `MENU_ICON` unless there is another specification besides XPM that is supported in the menu .   I read the freedesktop.org spec and this file doesn't seem to match, so please advise. Last, please group both `LINUX_DESKTOP_FOO` lines adjacent to each-other for organizational purposes.

Israel- replied:
Will this work?
`SET(LINUX_MENU_ICON "${DATA_DIR}/pixmaps/lmms.xpm")`

tresf replied:
The name is better, but unfortunately but `${DATA_DIR}/` is relative to our source/build directories which won't be of much value for repackaging.  We luck out by allowing the other icon to simply be `lmms` (assumes it is registered elsewhere in the system), but I'd like to know more about this menu in general, perhaps we have no easy way to configure it... or perhaps we just hard-code in the most sane spot that it `should be` installed? :confused:

tresf commented:
After Toby's comments on the mailing list, I understand why these changes had to be removed... Here's his feedback for those not on the mailing list:

> helper scripts are no problem as long as they're not within the debian directory. This directory should not exist for avoiding conflicts as the diff.gz of debian packages should contain a patch which creates all required files. If e.g. both upstream and Debian packagers modify the package and thus the changelog, it's likely there will be actual conflicts. So after all, do not touch packagers work and everything will be fine ;-)

I've made some line-comments on the updated commit.  Once these items are addressed, we can merge.

tresf commented:
@Israel- can you rebase and squash this commit please?

Israel- commented:
sorry @tresf  I have been soooooo busy lately.
I will have to start over.
My computer died :(  But, fortunately I have another, and will have to do this over.
I will try to get to it soon, but I am very busy.  I have not forgotten you :)

tresf commented:
@Israel- closing out.  Please issue a new PR when you have time.
